python/pyarrow/pandas_compat.py

```diff
@@ -470,7 +470,7 @@ def _get_range_index_descriptor(level):
     # public start/stop/step attributes added in pandas 0.25.0
     return {
         'kind': 'range',
-        'name': level.name,
+        'name': str(level.name) if level.name else None,
```

Comment:
Or perhaps `name` should be coerced to a string on `RangeIndex` instantiation? Perhaps there are other places where a `name` that's something like `np.int64(6)` might be problematic.

Reply:
pandas allows `Index.name` to be anything -- we may want to make our best effort to try to preserve the type of the name, in the worst case pickling could be necessary

Reply:
I see, that context helps, thanks! I hear `name` and immediately assume a string is the end goal.

I've amended the commit to at least preserve the type if default serializable. I'll noodle some more on pickling.